skrub/_dispatch.py

```
@@ -0,0 +1,194 @@
+...     print("calling pandas Series implementation")
+...     print("               ^^^^^^               ")
+
+>>> my_function(pd.Series([0]))
+calling pandas Series implementation
+               ^^^^^^
+>>> my_function(pd.DataFrame())
+calling pandas implementation
+
+Note that the last registered implementation wins -- there is no magic to
+prioritize specializations based on how specific they are.
+
+>>> @my_function.specialize("pandas")
+... def _override_all_pandas_specializations(df, other_arg=1.0):
+...     print("Any pandas (Series or DataFrame)")
+
+>>> my_function(pd.Series([0]))
+Any pandas (Series or DataFrame)
+>>> my_function(pd.DataFrame())
+Any pandas (Series or DataFrame)
+
+Importantly, single dispatch is done based on the type of the
+**first argument**. This means our generic function should be written in such a way
+that the first argument is the dataframe or column that will determine which
+implementation to use. For example the following would not work:
+
+>>> @dispatch
+... def sample(n_rows, df):
+...     ...
+
+Instead we must rewrite it to be:
+
+>>> @dispatch
+... def sample(df, n_rows):
+...     ...
+"""
+
+from functools import singledispatch
+
+
+def _load_dataframe_module_info(name):
+    # if the module is not installed, import errors are propagated
+    if name == "pandas":
+        import pandas
+
+        return {
+            "module": pandas,
+            "types": {
+                "DataFrame": [pandas.DataFrame],
+                "Column": [pandas.Series],
+            },
+        }
+    if name == "polars":
+        import polars
+
+        return {
+            "module": polars,
+            "types": {
+                "DataFrame": [polars.DataFrame, polars.LazyFrame],
+                "LazyFrame": [polars.LazyFrame],
+                "EagerFrame": [polars.DataFrame],
+                "Column": [polars.Series],
+            },
+        }
+    raise KeyError(f"Unknown module: {name}")
```

We should probably raise an error with the available backend.